tests/operations.rs

```diff
@@ -75,6 +76,77 @@ fn push_push_add() {
     run_program_assert_result(program, a + b);
 }
 
+#[test]
+fn dup1_once() {
+    let program = vec![
+        Operation::Push32(new_32_byte_immediate(10)),
+        Operation::Push32(new_32_byte_immediate(32)),
+        Operation::DupN(1),
+    ];
```

Comment:
We should add a `Pop` just for good measure.
```suggestion
        Operation::DupN(1),
        Operation::Pop,
    ];
```